dotnet-peer/Libp2pChat/Infrastructure/Libp2p/Libp2pChatService.cs

```diff
@@ -0,0 +1,342 @@
+    private bool _isStarted;
+    private bool _isDisposed;
+    private readonly string _listenAddress;
+    
+    /// <inheritdoc />
+    public string PeerId => _identity.PeerId.ToString();
+    
+    /// <inheritdoc />
+    public event EventHandler<ChatMessage>? MessageReceived;
+    
+    /// <inheritdoc />
+    public event EventHandler<Peer>? PeerDiscovered;
+    
+    /// <summary>
+    /// Creates a new instance of the <see cref="Libp2pChatService"/> class.
+    /// </summary>
+    /// <param name="logger">The application logger.</param>
+    /// <param name="peerFactory">The peer factory.</param>
+    /// <param name="router">The pubsub router.</param>
+    /// <param name="topicName">The topic name to use for chat messages.</param>
+    /// <param name="identity">The identity to use. If null, a new identity will be generated.</param>
+    public Libp2pChatService(
+        IAppLogger logger,
+        IPeerFactory peerFactory,
+        PubsubRouter router,
+        string topicName = "universal-connectivity",
+        Identity? identity = null)
+    {
+        _logger = logger;
+        _peerFactory = peerFactory;
+        _router = router;
+        _topicName = topicName;
+        _identity = identity ?? new Identity();
+        _listenAddress = "0.0.0.0";
+    }
+    
+    /// <inheritdoc />
+    public string GetMultiaddress()
+    {
+        return $"/ip4/{_listenAddress}/tcp/9096/p2p/{PeerId}";
+    }
+    
+    /// <inheritdoc />
+    public IReadOnlyCollection<Peer> GetKnownPeers()
+    {
+        return _knownPeers.Values.ToList().AsReadOnly();
+    }
+    
+    /// <inheritdoc />
+    public async Task SendMessageAsync(string message)
+    {
+        if (!_isStarted)
+            throw new InvalidOperationException("The chat service has not been started.");
+        
+        if (_topic == null)
+            throw new InvalidOperationException("The chat topic is not available.");
+            
+        try
+        {
+            // Send as plain text for compatibility with other implementations
+            byte[] data = Encoding.UTF8.GetBytes(message);
+            _topic.Publish(data);
+            
+            _logger.LogDebug($"Message sent: {message}");
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError("Failed to send message", ex);
+            throw;
+        }
+    }
+    
+    /// <inheritdoc />
+    public async Task StartAsync(CancellationToken cancellationToken)
+    {
+        if (_isStarted)
+            return;
+            
+        _logger.LogInformation("Starting chat service...");
+        _cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+        
+        try
+        {
+            // Create local peer
+            _localPeer = _peerFactory.Create(_identity);
+            
+            // Create multiaddresses to listen on
+            var localAddress = Multiaddress.Decode($"/ip4/127.0.0.1/tcp/9096/p2p/{PeerId}");
```

Comment:
Is 9096 a fixed port? Can it be parametrized? Or random. To listen to random one you can pass 0 instead